debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java

```diff
@@ -327,16 +330,16 @@ public void dispatchSchemaChangeEvent(Collection<T> dataCollectionIds, SchemaCha
         schemaChangeEventEmitter.emitSchemaChangeEvent(new SchemaChangeEventReceiver());
     }
 
-    public void alwaysDispatchHeartbeatEvent(OffsetContext offset) throws InterruptedException {
+    public void alwaysDispatchHeartbeatEvent(Partition partition, OffsetContext offset) throws InterruptedException {
         heartbeat.forcedBeat(
-                offset.getPartition(),
+                partition.getSourcePartition(),
```

Comment:
Sounds good. Let me try.

Reply:
From the TODO note, I understand that `OffsetContext` cannot be used here instead of `Map<String, ?>`:
https://github.com/debezium/debezium/blob/5c6750031393c104e1deaf3ccd8aac6d23924581/debezium-core/src/main/java/io/debezium/heartbeat/Heartbeat.java#L114-L115

For the sake of consistency, I'd leave `Map<String, ?>` instead of `Partition` here as well unless this TODO is no longer relevant.

Reply:
Ah, I see. So this relates to the legacy implementation for MySQL (i.e. not based on the common connector framework), which is supposed to be removed soon. I'm ok with passing the map-based representation for now.